File: data/multilang_comparison/program_c/src/func28.c
#include "program_c.h" 
array_t* func28(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array7 = func29(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array7->id);
      free(params0.data);
      array7->refC--;
      if(array7->refC == 0) {
         free(array7->data);
         free(array7);
         DEBUG_FREE(array7->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array15 = func30(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array15->id);
      free(params0.data);
      array15->refC--;
      if(array15->refC == 0) {
         free(array15->data);
         free(array15);
         DEBUG_FREE(array15->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array27 = func31(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array27->id);
   free(params0.data);
   return array27;
}

File: data/multilang_comparison/program_nim/src/func28.nim
proc func28*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array7: ptr ArrayT = func29(addr params0, getPath(), loopsFactor)
      debugReturn(array7.id)
      dealloc(params0.data)
      dec array7.refC
      if array7.refC == 0:
         dealloc(array7.data)
         dealloc(array7)
         debugFree(array7.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array68: ptr ArrayT = func30(addr params0, getPath(), loopsFactor)
      debugReturn(array68.id)
      dealloc(params0.data)
      dec array68.refC
      if array68.refC == 0:
         dealloc(array68.data)
         dealloc(array68)
         debugFree(array68.id)
   var params0: ArrayTParam
   params0.size = 0
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   var array104: ptr ArrayT = func31(addr params0, getPath(), loopsFactor)
   debugReturn(array104.id)
   dealloc(params0.data)
   return array104

